### ISSUE TITLE
Listen to comm messages on comms opened by the kernel

### ISSUE THREAD

**Issue by sbrunk** [almond-sh/almond#384]
I’m trying to figure out if it's feasible to port the ipywidgets kernel side to almond so I'm wrapping my head around comms. 

For ipywidgets I have to initiate a comm from the kernel and then listen to frontend messages of that comm (meaning with that comm id).

I can easily open a comm on the server and send messages
```scala
val comm = kernel.comm.sender("my_target", …)
comm.message("...")
```
But I can't find a way to register a callback or something similar for messages from the frontend on that existing comm. Am I missing something obvious here or is it not implemented yet?

For comparison, here's how you would do that in Python.
```python
from ipykernel.comm import Comm

# Use comm to send a message from the kernel
my_comm = Comm(target_name='my_comm_target', data={'foo': 1})
my_comm.send({'foo': 2})

# Add a callback for received messages.
@my_comm.on_msg
  def _recv(msg):
    ...
```

**Comment by alexarchambault** [comment]
[This example](https://nbviewer.jupyter.org/gist/alexarchambault/8772492819e2c7f9f256d497f2eea8c5) works from Jupyter classic.

**Comment by alexarchambault** [comment]
Seems it's what you asked actually 😅

**Comment by sbrunk** [comment]
Yes kernel -> frontend works fine. Frontend -> kernel works if you open the comm from the **frontend** and register a target in the kernel as in your example.

But if you open the comm from the **kernel**, frontend -> kernel doesn't seem to work. I'm not sure how to actually register a callback in that case...

I tried adding a receiver with the same target name like so but to set a callback:

```scala
val comm = kernel.comm.sender("my_target", …)
comm.message("...")
kernel.comm.receiver("my_target") { data => someCallback(...) }
```

But it seems that's just listening for a `comm_open` message for "my_target", not for regular comm messages with the already existing comm's id.

Perhaps what's missing is the registration of a target with the comm id when we open the comm on the kernel side. At least that's happening in `commOpenHandler`, i.e. when we receive a `comm_open` message:

https://github.com/almond-sh/almond/blob/3e4c5ea6815d964e0b663b2746f78d5b85443e0a/modules/shared/interpreter/src/main/scala/almond/interpreter/messagehandlers/CommMessageHandlers.scala#L26-L28

**Comment by sbrunk** [comment]
Implemented in #389
